tests/js/common/shell/shell-validation-rocksdb.js

```diff
@@ -360,6 +360,28 @@ function ValidationBasicsSuite () {
       }
 
     },
+    // AQL  ////////////////////////////////////////////////////////////////////////////////////////////
+    testGET_SCHEMA: () => {
+      const v =  validatorJson;
+      v.level = "strict";
+      testCollection.properties({"validation" : v });
+      sleepInCluster();
+
+      let res = db._query(`RETURN GET_SCHEMA("${testCollectionName}")`).toArray();
+      assertEqual(res[0], v.rule);
+    },
+
+    testSCHEMA_VALIDATE: () => {
+      const v =  validatorJson;
+      testCollection.properties({"validation" : { rule:{} } });
+      sleepInCluster();
+
+      let res = db._query(`RETURN SCHEMA_VALIDATE({"foo" : 24}, { "properties" : { "foo" : { "type" : "string" } } } )`).toArray();
+      assertEqual(res[0].valid, false);
+
+      res = db._query(`RETURN SCHEMA_VALIDATE({"foo" : "bar"}, { "properties" : { "foo" : { "type" : "string" } } } )`).toArray();
+      assertEqual(res[0].valid, true);
+    },
```

Comment:
We should have tests for several additional cases:
* invalid schemas
* documents that do not match the schema
* more complex schemas
* using `null` as the schema (e.g. because a return value of `GET_SCHEMA` is used unchecked)

Reply:
Added more tests:

This schema is used in test and it is complex enough to test all of the ArangoDB functionality. In the first versions `additionalProperties` was not tested as I must be. But now the tests checks if attributes like `_key`, `_id` etc are added or not. For the rest we must rely on the tests of the underlying lib.

```
        rule : {
          type: "object",
          properties: {
            zahlen : {
              type : "array",
              items : { type : "number", maximum : 6 }
            },
            name : {
              type : "string",
              minLength: 4,
              maxLength: 10
            },
            number : {
              type : "number",
              items : { minimum: 1000000 }
            },
          },
          additionalProperties : false
        },

```